cuckoo/table.go

```diff
@@ -77,22 +38,22 @@ type Table struct {
 // numBuckets = number of buckets
 // depth = the number of entries per bucket
 // randSeed = seed for PRNG
-func NewTable(name string, numBuckets int, bucketDepth int, itemSize int,
+func NewTable(name string, numBuckets uint64, bucketDepth uint64, itemSize uint64,
 	data []byte, randSeed int64) *Table {
 	t := &Table{name, numBuckets, bucketDepth, itemSize, nil, nil, nil, nil}
-	t.log = log.New(os.Stderr, "[Cuckoo:"+name+"] ", log.Ldate|log.Ltime|log.Lshortfile)
-	t.rand = rand.New(rand.NewSource(randSeed))
-
 	if data == nil {
 		data = make([]byte, numBuckets*bucketDepth*itemSize)
 	}
-	if len(data) != numBuckets*bucketDepth*itemSize {
-		return nil
-	}
 	t.data = data
-
+	t.rand = rand.New(rand.NewSource(randSeed))
```

Comment:
Cuckoo.insert is already the bottleneck for coordinator. Would rather not if not necessary. Don't think there is any need anyway, there's nothing about the cuckoo layout that needs to be secret